### ISSUE TITLE
For MAF: upload SNVs of normal samples to Genotype

### ISSUE THREAD

**Issue by moahaegglund** [Clinical-Genomics/BALSAMIC#882]
## Description

When a tumor/normal analysis is ordered with BALSAMIC analysis only we need to create a case for the MIP analysis of the normal sample under cust000 to compare with the results of MAF.  For example the samples in ticket 110945. Today prod bioinfo have to create these cases manually.

**Comment by moahaegglund** [comment]
@ashwini06 would it be possible to upload the SNVs of the normal samples to Genotype?

**Comment by ashwini06** [comment]
@moahaegglund @henrikstranneheim: Somehow I missed this issue to answer, sorry about that! Yes, it might be possible. Currently, BALSAMIC reports SNV germline variants from normal samples. We have those results generated from two separate variants callers (Sentieons DNAscope and GATK haplotypecaller). Maybe we can use one of those for genotype?

**Comment by Vince-janv** [comment]
I tested uploading `/home/proj/production/housekeeper-bundles/manydoe/2022-04-26/SNV.germline.normal.dnascope.vcf.gz` to genotype-stage. Using the match sequence function genotype then matched it against the vcf uploaded from the mip-analysis.

@ashwini06 However genotype gets the sample name from the VCF header and in BALSAMIC the field is called `NORMAL` for the normal sample. Can this easily be changed to the sample internal_id?

**Comment by ashwini06** [comment]
@Vince-janv : 

> I tested uploading `/home/proj/production/housekeeper-bundles/manydoe/2022-04-26/SNV.germline.normal.dnascope.vcf.gz` to genotype-stage. Using the match sequence function genotype then matched it against the vcf uploaded from the mip-analysis.
> 
> @ashwini06 However genotype gets the sample name from the VCF header and in BALSAMIC the field is called `NORMAL` for the normal sample. Can this easily be changed to the sample internal_id?

@Vince-janv : Yes, it is possible but currently we don't have the functionality to make this change. But with a simple command-line, one can easily change the sample names in the output VCF files.


 `bcftools reheader -s change_sample_names.txt SNV.germline.normal.dnascope.vcf.gz -o  SNV.germline.normal.dnascope.changednamed.vcf.gz` 

`change_sample_names.txt` is a 2-column text file. For eg: 
NORMAL ACCxxxx

So it is only you are interested in the sample name change in this output VCF : SNV.germline.**normal**.dnascope.vcf.gz?

**Comment by Vince-janv** [comment]
@ashwini06 Thanks for the quick reply! 😊 I think there is an argument for consistency (ie. using the same header format in all VCFs). But that being said I know very little about how BALSAMIC uses the VCF headers. But for genotype we just need a VCF for the normal, and i chose one of the ones you suggested.

The "manual" change you suggested will be great for testing but for use in production we would need another solution.

**Comment by Vince-janv** [comment]
@ivadym we would also need to add the tag `genotype` to the germline DNAscope file in housekeeper 🙇‍♂️

**Comment by Vince-janv** [comment]
Hi. I will summarise some diskussions we've the past weeks.

@ashwini06 Informed me that changing the vcf headers is more problematic than initially thought.

Discussing with @Mropat and @ivadym we concluded that implementing changes in genotype would be beneficial, but would take a lot of work. With the current lack of developers I think it might not happen for a while if not prioritised.

As a quick and dirty fix I suggest that @ivadym adds the tag in housekeeper and I'll merge a small PR in cg to adapt the current genotype upload to balsamic. Production can then re-header the file manually and upload via the cg CLI.

Let me stress that this should not be a permanent solution!

**Comment by ashwini06** [comment]
@Vince-janv : Thanks for the update! I have two possible solutions that might work, but again I am on it and testing before I promise something to you.
a) Within BALSAMIC, Just renaming the VCF headers of genotype normal VCF file and saving the file specifically for genotype. This way we are only creating an extra VCF file for genotype without disturbing the format of sample names in the rest of the output VCF files and BAM files. 

b) To have the same standardized format for sample names (i.e LIMSID instead of TUMOR/NORMAL) in all output VCF and BAM files, it requires huge refactoring with most of the BALSAMIC snakemake rules (for [eg](https://github.com/Clinical-Genomics/BALSAMIC/blob/2e7ef3d5d437424b0dc9b75253152fbc90a92fbf/BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_normal.rule#L75)). This could be a permanent solution. This also need some changes on SCOUT balsamic config file creation i guess.. for example: the following `sample_id` needs to set to LIMSID. @ivadym : can you confirm me if it is done automatically based on the cram file header or someother way?

```
samples:
- alignment_path: /home/proj/stage/cancer/cases/sweetelf_test_APJ/analysis/bam/tumor.merged.cram
  analysis_type: panel
  phenotype: affected
  sample_id: TUMOR
  sample_name: GMSmyeloid-control-HD829-1
  sex: unknown
  tissue_type: cell line
  tumor_purity: 0
track: cancer
```
```
samples:
- alignment_path: /home/proj/stage/cancer/cases/sweetelf_test_APJ/analysis/bam/tumor_namechanged.merged.bam
  analysis_type: panel
  phenotype: affected
  sample_id: ACC6637A1
  sample_name: GMSmyeloid-control-HD829-1
  sex: unknown
  tissue_type: cell line
  tumor_purity: 0
track: cancer
```

@Vince-janv @ivadym I am on my way in testing my solutions. if its OK, can i ask both to wait a bit before you start fixing your proposed solutions in cg and HK.

**Comment by ivadym** [comment]
@ashwini06 The default values for `sample_id` are actually the `lims_id`, for BALSAMIC this is being changed in CG to tumor/normal. So its independent of the cram files, but I will look more in detail what would be affected by this changes.

**Comment by ashwini06** [comment]
Fixed in #958
